src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1294,8 +1294,9 @@ internal static NamedTypeSymbol GetIEnumerableOfT(TypeSymbol type, bool isAsync,
             if (type.TypeKind == TypeKind.TypeParameter)
             {
                 var typeParameter = (TypeParameterSymbol)type;
-                GetIEnumerableOfT(typeParameter.EffectiveBaseClass(ref useSiteDiagnostics).AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics), isAsync, compilation, ref @implementedIEnumerable, ref foundMultiple);
-                GetIEnumerableOfT(typeParameter.AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics), isAsync, compilation, ref @implementedIEnumerable, ref foundMultiple);
+                var allInterfaces = typeParameter.EffectiveBaseClass(ref useSiteDiagnostics).AllInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics)
+                    .Concat(typeParameter.AllEffectiveInterfacesWithDefinitionUseSiteDiagnostics(ref useSiteDiagnostics));
```

Comment:
>Concat [](start = 21, length = 6)

I think we have Concat for ImmutableArrays that return an ImmutableArray.

Reply:
Yes, I thought it was returning an IEnumerable, but it's actually returning an ImmutableArray. Nice

---
In reply to: [321522503](https://github.com/dotnet/roslyn/pull/38460#discussion_r321522503) [](ancestors = 321522503)